### ISSUE TITLE
Warnings when pycache files are found in a provider

### ISSUE THREAD

**Issue by pieqq** [canonical/checkbox#565]
Since #545 was merged, we see this kind of warnings in Checkbox (this example is from when Checkbox is installed via the Debian packages):

```
checkbox-cli
=========================[ Resume Incomplete Session ]==========================
There are 0 incomplete sessions that might be resumed
Preparing...
WARNING:plainbox.secure.providers.v1:Skipped file: /usr/lib/checkbox-provider-base/bin/__pycache__/gateway_ping_test.cpython-310.pyc
```

This kind of files should be ignored, not reported.

**Comment by kissiel** [comment]
I guess we could approach this problem with either warn_list or ignore_list. What I mean is quietly ignore stuff that match patterns on ignore_list or present warning only for files that match patterns on warn_list.
